internal/cmd/kafka/command_mirror_list.go

```diff
@@ -55,53 +56,52 @@ func (c *mirrorCommand) list(cmd *cobra.Command, _ []string) error {
 	}
 
 	kafkaREST, err := c.GetKafkaREST()
-	if kafkaREST == nil {
-		if err != nil {
-			return err
-		}
-		return errors.New(errors.RestProxyNotAvailableMsg)
+	if err != nil {
+		return err
 	}
 
 	cluster, err := c.Context.GetKafkaClusterForCommand()
 	if err != nil {
 		return err
 	}
 
-	mirrorStatusOpt := optional.EmptyInterface()
+	var mirrorTopicStatus *kafkarestv3.MirrorTopicStatus
 	if mirrorStatus != "" {
-		mirrorStatusOpt = optional.NewInterface(kafkarestv3.MirrorTopicStatus(mirrorStatus))
+		mirrorTopicStatus, err = kafkarestv3.NewMirrorTopicStatusFromValue(strings.ToUpper(mirrorStatus))
```

Comment:
The `kafka-rest-sdk-go` version accepted lowercase strings here, while the `ccloud-sdk-go-v2` version does not. I added `strings.ToUpper` here so that this doesn't cause a breaking change, but we could remove this in V4 and enforce uppercase.